doc/build-helpers/fetchers.chapter.md

```diff
@@ -70,28 +126,32 @@ fetchurl {
 }
 ```
 
-When building the recipe, you'll get an error message that you can use to determine the correct hash:
+When building the package, use the error message to determine the correct hash:
 
 ```shell
 $ nix-build
-(some output removed for clarity)
+...
 error: hash mismatch in fixed-output derivation '/nix/store/7yynn53jpc93l76z9zdjj4xdxgynawcw-version.drv':
          specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
             got:    sha256-BZqI7r0MNP29yGH5+yW2tjU9OOpOCEvwWKrWCv5CQ0I=
 error: build of '/nix/store/bqdjcw5ij5ymfbm41dq230chk9hdhqff-version.drv' failed
 ```
+:::
 
-[]{#fetchers-hash-tip}
-:::{.tip}
-For most cases, it is recommended to use the method illustrated above to figure out the appropriate hash to use in a fetcher:
-set the `hash` attribute to an empty string, try to build the derivation, and then copy the hash given by the error message.
+## Obtaining hashes securely {#sec-pkgs-fetchers-secure-hashes}
 
-Remember to set the `hash` attribute to an empty string whenever you update fetcher parameters and expect the content to change!
-:::
+Let's say Man-in-the-Middle (MITM) sits close to your network.
+Then instead of fetching source you can fetch malware, and instead of the actual source hash, you get the hash of malware.
+Here are security considerations for this scenario:
 
-A similar problem arises while testing changes to a fetcher's implementation.
-If the output of the derivation already exists in the Nix store, test failures can go undetected.
-The [`invalidateFetcherByDrvHash`](#tester-invalidateFetcherByDrvHash) function helps prevent reusing cached derivations.
+- `http://` URLs are not secure to prefetch hashes.
+
+- Upstream hashes should be obtained via a secure protocol.
+
+- `https://` URLs are secure when using `nix-prefetch-*` or for upstream hashes.
+
+- `https://` URLs are secure relying on "trust on first use" *only if* you use one of the listed fake hashes.
+  If you use any other hash, `fetchurl` will pass `--insecure` to `curl` and may then degrade to HTTP in case of TLS certificate expiration.
```

Comment:
Mentioning `fetchurl` details will feel confusing here. If this is new information for anyone reading this, chances are very low that they will understand what "`fetchurl` will pass `--insecure` to `curl`" and what "TLS certificate expiration" means.

I'm adapting this to be most understandable to the audience which will find this section relevant.

```suggestion
- `https://` URLs are secure when using the [fake hash method](#sec-pkgs-fetchers-updating-source-hashes-fakehash-method) *only if* you use one of the listed fake hashes.
  If you use any other hash, the download will be exposed to MITM attacks even if you use HTTPS URLs.
```

Reply:
No, security-related information should be concrete. This smudges away the facts. We can be a bit more clear and say

> will pass `--insecure` to the underlying call to` curl`

Reply:
Added this information, I think it covers all audiences now.